fixtures/helpers.js

```diff
@@ -0,0 +1,104 @@
+import * as ReactDOM from "react-dom";
+
+export async function sleep(duration = 0) {
+    return new Promise((resolve, reject) => {
+        setTimeout(resolve, duration);
+    });
+}
+
+export async function simulate(event) {
+    await fetch("http://localhost:3000/simulate", {
+        method: "POST",
+        headers: {
+            Accept: "application/json",
+            "Content-Type": "application/json",
+        },
+        body: JSON.stringify({
+            offsetX: window.screenX,
+            offsetY: window.screenY + (window.outerHeight - window.innerHeight),
+            event: event,
+        }),
+    });
+}
+
+let container;
+
+export async function render(element) {
+    return new Promise((resolve, reject) => {
+        ReactDOM.render(element, container, () => {
+            setTimeout(() => resolve(container), 0);
+        });
+    });
+}
+
+let currentTest = null;
+
+export async function runTests() {
+    for (const test of tests) {
+        currentTest = test;
+        container = document.createElement("div");
+        container.style.display = "inline-block";
+        document.body.appendChild(container);
+        await test.run();
+        ReactDOM.unmountComponentAtNode(container);
+        document.body.removeChild(container);
+        await sleep();
+    }
+}
+
+const counters = {};
+
+export function expect(domElement) {
+    return {
+        toMatchScreenshot: async (options = {}) => {
+            const bounds = domElement.getBoundingClientRect();
+            const titlebarHeight = window.outerHeight - window.innerHeight;
+
+            const outset = options.outset || 0;
+            const title = currentTest.title
+                .replace(/, /g, "_")
+                .replace(" ", "_");
```

Comment:
I should probably keep title as an array of strings and `join` then using appropriate separators where needed.